### ISSUE TITLE
Latest commit of tlib breaks snipmate

### ISSUE THREAD

**Issue by marcusboon** [tomtom/tlib_vim#24]
Updated tlib to version 1.23 and it breaks snipmate with a
`Error detected while processing /home/marcus/.vim/bundle/vim-snipmate/autoload/snipMate.vim:
line    9:
you're missing tlib. See install instructions at /home/marcus/.vim/bundle/vim-snipmate/README.md
Press ENTER or type command to continue`

Reverting back to 1.22 solves the issue.

**Comment by craigmaloney** [comment]
Reason being is tlib 1.23 has a version check in World.vim for Vim 8.0. So it looks like vim 7 is no longer working with the latest change.

**Comment by craigmaloney** [comment]
`Error detected while processing /home/craig/configs/vim/bundle/tlib_vim/autoload/tlib/World.vim:
line    8:
tlib: VIM >= 8.00 is required
Error detected while processing function tlib#input#List:
line   16:
`

**Comment by tomtom** [comment]
Well yes, tlib >= 1.23 requires VIM 8 or higher. The reason is that it now makes use of the new winid-related functions. I could try to imitate this behaviour for VIM 7 but this isn't a high-priority goal for me.

Version 1.22 should still work well in conjunction with snipmate.

Are you stuck with vim 7?

**Comment by craigmaloney** [comment]
Until I can upgrade to a later version of Ubuntu (currently at 14.04) I'm stuck at Vim 7.4. I believe this is the case for 16.04 as it doesn't ship with Vim 8 and there's no PPA support that I'm aware of.

**Comment by taq** [comment]
This is affecting also Neovim users, as Neovim is returning the `704` version with `v:version`. I fixed it checking if we are really using Vim and checking for version like this:

```
if v:progname == 'vim' && v:version < 800
    echoerr 'tlib: VIM >= 8.00 is required'
    finish
endif
```

Can we use that way or is there some better way to do that?

**Comment by tomtom** [comment]
> This is affecting also Neovim users

Neovim doesn't support lambdas either, does it?

**Comment by craigmaloney** [comment]
Thank you. Verified that this works again.

**Comment by taq** [comment]
@tomtom seems that it does not support lambdas - yet - but are working on it.

**Comment by yann-morin-1998** [comment]
Indeed, this fixes garbas/vim-snipmate#246

Thanks!
